eng/pipelines/templates/steps/build-test.yml

```diff
@@ -4,67 +4,94 @@ parameters:
+      foreach ($md in $modDirs) {
+        pushd $md
+        Write-Host "##[command]Executing go vet ./... in $md"
+        go vet ./...
+      }
     displayName: 'Vet'
     workingDirectory: '${{parameters.GoWorkspace}}'
     env:
       GO111MODULE: 'on'
 
-  - ${{ if eq(parameters.RunTests, 'true') }}:
-    - pwsh: |
-        go get github.com/jstemmer/go-junit-report
-        go get github.com/axw/gocov/gocov
-        go get github.com/AlekSi/gocov-xml
-        go get github.com/matm/gocov-html
-        go get -u github.com/wadey/gocovmerge
-      displayName: "Install Coverage and Junit Dependencies"
-      workingDirectory: '${{parameters.GoWorkspace}}'
+  - template: configure-proxy.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
 
-    - pwsh: |
-        $testDirs = (./eng/scripts/get_test_dirs.ps1 -serviceDir $(SCOPE))
-        foreach ($td in $testDirs) {
-          pushd $td
-          Write-Host "##[command]Executing 'go test -run "^Test" -v -coverprofile coverage.txt .' in $td"
-          go test -run "^Test" -v -coverprofile coverage.txt . | Tee-Object -FilePath outfile.txt
-          if ($LASTEXITCODE) { exit $LASTEXITCODE }
-          cat outfile.txt | go-junit-report > report.xml
-          # if no tests were actually run (e.g. examples) delete the coverage file so it's omitted from the coverage report
-          if (Select-String -path ./report.xml -pattern '<testsuites></testsuites>' -simplematch -quiet) {
-            Write-Host "##[command]Deleting empty coverage file"
-            rm coverage.txt
-          }
+  - pwsh: |
+      $testDirs = (./eng/scripts/get_test_dirs.ps1 -serviceDir $(SCOPE))
+      foreach ($td in $testDirs) {
+        pushd $td
+        $(Build.SourcesDirectory)/eng/scripts/start-server.ps1 start
+        Write-Host "##[command]Executing 'go test -run "^Test" -v -coverprofile coverage.txt .' in $td"
+        go test -run "^Test" -v -coverprofile coverage.txt . | Tee-Object -FilePath outfile.txt
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+        cat outfile.txt | go-junit-report > report.xml
+        # if no tests were actually run (e.g. examples) delete the coverage file so it's omitted from the coverage report
+        if (Select-String -path ./report.xml -pattern '<testsuites></testsuites>' -simplematch -quiet) {
+          Write-Host "##[command]Deleting empty coverage file"
+          rm coverage.txt
         }
-      displayName: 'Run Tests'
-      workingDirectory: '${{parameters.GoWorkspace}}'
-      env:
-        GO111MODULE: 'on'
-
-    - pwsh: ../eng/scripts/create_coverage.ps1 ${{parameters.ServiceDirectory}}
-      displayName: 'Generate Coverage XML'
-      workingDirectory: '${{parameters.GoWorkspace}}sdk'
+      }
+      $(Build.SourcesDirectory)/eng/scripts/start-server.ps1 stop
+    displayName: 'Run Tests'
+    workingDirectory: '${{parameters.GoWorkspace}}'
+    env:
+      GO111MODULE: 'on'
+      AZURE_RECORD_MODE: 'playback'
```

Comment:
It would be good to have this as a parameter that defaults to `playback` so that when we add live tests we can pass in `live` from the calling template.

Reply:
But we can also just do that work when we get to it.